### DISCUSSION TITLE
Link and useRouter() not working with client side and it reloads the page - Next Js

### DISCUSSION THREAD

**Discussion by mvn-tinhle-dn** [vercel/next.js#62708] [Help]
### Summary

I use Link in NextJs and am having problem with it, the problem is as follows:
My link doesn't work with client-side and always reloads the page with the url as `"/app/templates/application_form"`. Just change the url `"/app/templates/application_form"` to `"/app/templates/application_forms"` or something else and it works fine, normally and without any problems.
I tried searching in my project to find a duplicate with the above url but there is no duplicate here.
Even when I use this url with `router.push("/app/templates/application_forms")` it has the same problem, while other urls don't have it at all.
Hope someone can help me. 
Thanks!

### Additional information

```js
next: 13.4.12
```

### Example

import { useRouter } from "next/router";
import Link from "next/link";

const Test= () => {
    const router = useRouter();
....
return (
....
    `<Link href="/app/templates/application_form">Application form</Link>`
    `<button onClick={() => router.push("/app/templates/application_forms") } >go to Application form</button>`
....
)
}

**Comment by icyJoseph** [comment]
Two things, first, why are you prefixing the links with `/app`? Are you using App router and have `app/app/templates/application_form/page.tsx`?

Second, if you are using App router, you have to import `useRouter` from `'next/navigation'`

Share your application file structure, so that we can help more.

**Comment by Gali-Madhan-Kumar** [comment]
Hi, in next js useRouter() not working as expected navigate.push(path) changing url but page is not refreshing please let me know what can be the issue

import React, { useEffect, useState } from "react";
import {
  Grid,
  Button,
  List,
  ListItem,
  ListItemText,
  ListItemSecondaryAction,
  Typography,
  Tooltip,
  IconButton,
  Paper,
  Collapse,
  Box,
} from "@mui/material";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import Cookies from "js-cookie";
import classes from "./EmpHomePage.module.scss";
import CustomCard from "./Cards";
import CorporateFareOutlinedIcon from "@mui/icons-material/CorporateFareOutlined";
import {
  getPackagesDetails,
  getRecentActivities,
} from "../../service/employer/home";
import {
  ACCOUNT_TYPES,
  USER_TYPES,
  copyToClipBoard,
  formatDate,
  getFormattedJobTitle,
  scrollToTop,
} from "../../util/helper";
import { useRouter } from "next/router";
import BuyPackagePopup from "../../components/modal/BuyPackagePopup";
import { PLACEHOLDERS } from "../../util/Labels";
import ContentCopyOutlinedIcon from "@mui/icons-material/ContentCopyOutlined";
import colors from "../../vars.module.scss";
import { useTheme } from "@mui/system";
import useMediaQuery from "@mui/material/useMediaQuery";
import { useSelector, useDispatch } from "react-redux";
import InfoOutlinedIcon from "@mui/icons-material/InfoOutlined";
import QrCodeScannerOutlinedIcon from "@mui/icons-material/QrCodeScannerOutlined";
import {
  packageStatsDataError,
  packageStatsDataReceive,
  packageStatsDataRequest,
} from "../../redux/actions";
import WrapperTooltip from "../../components/common/ToolTip/Tooltip";
import { styled } from "@mui/system";
import PeopleIcon from "@mui/icons-material/People";
import FacebookIcon from "@mui/icons-material/Facebook";
import XIcon from "@mui/icons-material/X";
import LinkedInIcon from "@mui/icons-material/LinkedIn";
import EmailIcon from "@mui/icons-material/Email";
import WhatsAppIcon from "@mui/icons-material/WhatsApp";
import TelegramIcon from "@mui/icons-material/Telegram";
import InstagramIcon from "@mui/icons-material/Instagram";
import Head from "next/head";
import {
  Container,
  MainContent,
  SectionHeading,
} from "../../components/common/CommonStyles";
import SocialShare from "../../components/social-share/SocialShare";

const postAJobIcon = "/employer/assets/images/post_job.svg";
const manageJobIcon = "/employer/assets/images/manage_job.svg";
const talentSearchIcon = "/employer/assets/images/talent_search.svg";

export const CustomButton = styled(Button)(({ theme, height }) => ({
  height: `${height}em`,
  width: "100%",
  fontSize: "1.125em",
  color: colors?.primaryColor,
  border: `1px solid ${colors?.primaryColor}`,
  borderRadius: "8px",
  textTransform: "none",
  transition: "color 0.3s, transform 0.3s",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  margin: "0.5em",
  position: "relative",
  overflow: "hidden",
  zIndex: 1,
  "&::before": {
    content: '""',
    position: "absolute",
    top: 0,
    left: 0,
    width: "100%",
    height: "100%",
    borderRadius: "inherit",
    border: `2px solid ${colors?.primaryColor}`,
    boxSizing: "border-box",
    transform: "scaleX(0)",
    transformOrigin: "left",
    transition: "transform 0.3s",
    zIndex: -1,
  },
  "&:hover::before": {
    transform: "scaleX(1)",
  },
  "&:hover": {
    color: colors?.primaryColor,
    transform: "scale(1.05)",
    borderColor: colors?.primaryColor,
  },
  "&:active": {
    backgroundColor: theme.palette.action.selected,
    transform: "scale(1.05)", // Add a slight scale effect on active state for mobile
  },
  "&:focus": {
    outline: "none",
    boxShadow: `0 0 0 4px rgba(0,0,0,0.2)`,
  },
  "@media (max-width: 959px)": {
    fontSize: "1em",
    height: `${height - 1}em`,
    maxWidth: "90%", // Full width on small screens for better usability
    margin: "0.25em 0.4rem", // Reduce margin for better spacing
    "&:hover": {
      backgroundColor: theme.palette.action.selected, // Use active color on touch
      transform: "scale(1)", // Remove scale effect on hover for mobile
    },
    "&:active": {
      backgroundColor: theme.palette.action.selected,
      transform: "scale(1.05)", // Slight scale effect on active state for touch
    },
    "&:focus": {
      boxShadow: `0 0 0 4px ${theme.palette.primary.light}`,
    },
  },
}));

const CustomTypography = styled(Typography)(({ theme }) => ({
  color: colors?.primaryDarkColor,
  fontSize: "0.875em",
  fontWeight: 600,
  width: "82.5%",
  whiteSpace: "nowrap", // Prevent text from wrapping
  overflow: "hidden", // Hide any overflow content
  textOverflow: "ellipsis", // Display an ellipsis if the text is too long
  [theme.breakpoints.down("md")]: {
    width: "100%",
  },
}));

export const ExpandMore = styled((props) => {
  const { expand, ...other } = props;
  return <IconButton {...other} />;
})(({ theme, expand }) => ({
  transform: !expand ? "rotate(0deg)" : "rotate(180deg)",
  marginLeft: "auto",
  transition: theme.transitions.create("transform", {
    duration: theme.transitions.duration.shortest,
  }),
}));

const ListRightAction = styled("div")(({ theme, disabled }) => ({
  display: "flex",
  alignItems: "center",
  gap: theme.spacing(1), // Adjust gap as needed
  padding: `${theme.spacing(0.5)} ${theme.spacing(1)}`, // Smaller padding for mobile
  border: `1px solid ${
    disabled ? theme.palette.grey[400] : colors?.primaryColor
  }`, // Adjusted border color for disabled state
  borderRadius: theme.shape.borderRadius,
  cursor: disabled ? "default" : "pointer", // Disable cursor for disabled state
  transition: "background-color 0.3s, color 0.3s",
  backgroundColor: disabled ? theme.palette.grey[200] : "transparent", // Background color for disabled state
  color: disabled ? theme.palette.text.disabled : "inherit", // Text color for disabled state
  "&:hover": {
    backgroundColor: !disabled && colors?.primaryColor,
    color: !disabled && theme.palette.primary.contrastText,
    "& svg": {
      color: !disabled && theme.palette.primary.contrastText, // Change icon color on hover (for non-disabled state)
    },
  },
  [theme.breakpoints.down("md")]: {
    padding: `${theme.spacing(0.25)} ${theme.spacing(0.5)}`, // Adjusted padding for smaller screens
    gap: theme.spacing(0.5), // Adjusted gap for smaller screens
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
}));

const Icon = styled("span")({
  fontSize: "1.25em",
  color: colors?.primaryColor,
});

const Text = styled("span")({
  fontSize: "0.875em",
  color: (theme) => theme.palette.text.primary,
});

const SectionTitle = styled(Typography)(({ theme }) => ({
  fontSize: "1.5rem",
  fontWeight: "bold",
  textAlign: "center",
  margin: "1rem 0",
  color: colors?.primaryColor, // Use your primary color from theme
  "@media (max-width: 600px)": {
    fontSize: "1.2rem",
    marginBottom: 0,
  },
}));

export const ShareIconButton = styled(Button)(({ theme }) => ({
  margin: theme.spacing(1),
  padding: theme.spacing(1),
  backgroundColor: "transparent",
  color: colors?.primaryColor,
  border: "none",
  cursor: "pointer",
  "&:hover": {
    backgroundColor: theme.palette.action.hover,
  },
  "&:focus": {
    outline: "none",
  },
}));

const ReferralContainer = styled(Grid)(({ theme }) => ({
  padding: "1% 2%",
  [theme.breakpoints.down("sm")]: {
    flexDirection: "column",
    padding: "1% 5%",
  },
}));

const ReferralTextContainer = styled(Box)(({ theme }) => ({
  display: "flex",
  padding: "1% 0px",
  alignItems: "center",
  marginBottom: theme.spacing(2),
  [theme.breakpoints.up("md")]: {
    marginBottom: 0,
  },
}));

const ReferralLinkText = styled(Typography)(({ theme }) => ({
  fontSize: "0.75em",
  color: "black",
  wordBreak: "break-all",
  fontWeight: 500,
}));

const ReferralActions = styled(Grid)(({ theme }) => ({
  display: "flex",
  justifyContent: "flex-end",
  alignItems: "center",
  flexWrap: "wrap",
  "@media (max-width: 600px)": {
    justifyContent: "flex-start",
  },
}));

const EmpHomePage = () => {
  const theme = useTheme();
  const matches = useMediaQuery(theme.breakpoints.down("md"), { noSsr: true });
  const dispatch = useDispatch();
  const auth = useSelector((state) => state.auth);
  const packageData = useSelector((state) => state.package);
  const recentActivitiesData = useSelector((state) => state.recentActivities);
  const [recentActivities, setRecentActivities] = useState([]);
  const [openPackagePopup, setOpenPackagePopup] = useState(false);
  let navigate = useRouter();
  const [expanded, setExpanded] = React.useState(true);

  const empProfile = Cookies.get("empProfile")
    ? JSON.parse(Cookies.get("empProfile"))
    : null;

  const handleExpandClick = () => {
    scrollToTop();
    setExpanded(!expanded);
  };

  const referralLink = `${
    process.env.NEXT_PUBLIC_REACT_APP_DOMAIN
  }/employer/register${
    empProfile
      ? `?refcode=${empProfile?.publicId}&utm_campaign=Referral_Invite&utm_medium=Referral_Program&utm_source=share`
      : ""
  }`;

  const handleCopyLink = () => {
    copyToClipBoard(referralLink, setCopy);
  };

  const handleSocialShare = (platform) => {
    let shareUrl = referralLink;
    let shareText = `Sign up to discover top talent for your organization with JobsHorn!\n\n${shareUrl}\n\n#JobsHorn`;

    switch (platform) {
      case "facebook":
        window.open(
          `https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(
            shareUrl
          )}&quote=${encodeURIComponent(
            shareText
          )}&hashtag=${encodeURIComponent("#JobsHorn")}`,
          "_blank"
        );
        break;
      case "twitter":
        window.open(
          `https://twitter.com/intent/tweet?text=${encodeURIComponent(
            shareText
          )}`,
          "_blank"
        );
        break;
      case "linkedin":
        window.open(
          `https://www.linkedin.com/shareArticle?mini=true&url=${encodeURIComponent(
            shareUrl
          )}&text=${encodeURIComponent(shareText)}`,
          "_blank"
        );
        break;
      case "email":
        window.location.href = `mailto:?subject=${encodeURIComponent(
          "Sign up for amazing job opportunities at JobsHorn!"
        )}&body=${encodeURIComponent(shareText)}`;
        break;
      case "whatsapp":
        window.open(
          `https://api.whatsapp.com/send?text=${encodeURIComponent(shareText)}`,
          "_blank"
        );
        break;
      case "telegram":
        window.open(
          `https://telegram.me/share/url?url=${encodeURIComponent(shareText)}`,
          "_blank"
        );
        break;
      default:
        break;
    }
  };

  useEffect(() => {
    if (typeof window !== undefined) {
      if (!auth?.token) {
        window.location.href = "/employer/login";
      } else if (!auth?.hasAccess || !auth?.isVerified) {
        window.location.href = "/employer/register/employer-details";
      }
    }
  }, [auth?.hasAccess, auth?.isVerified, auth?.token]);

  useEffect(() => {
    if (empProfile) {
      if (empProfile?.companyProfileTypeCode !== ACCOUNT_TYPES.ADMIN) {
        if (!packageData?.data) {
          getPackages();
        }
        getActivities();
      }
    }
  }, []);

  const getPackages = async () => {
    dispatch(packageStatsDataRequest());
    try {
      const data = await getPackagesDetails(); // Use your API service function
      if (data !== null && data?.data) {
        dispatch(packageStatsDataReceive(data?.data));
      } else {
        throw new Error("API request failed");
      }
    } catch (error) {
      dispatch(packageStatsDataError(error));
    }
  };

  const getActivities = () => {
    try {
      const activities = getRecentActivities();
      activities.then((res) => {
        if (res?.status === 200 && res?.data) {
          setRecentActivities(res?.data);
        }
      });
    } catch (err) {
      console.log("Activities Api error ", err);
    }
  };

  const buyPackage = () => {
    setOpenPackagePopup(true);
    if (matches) {
      scrollToTop();
    }
  };

  const onClosePopup = () => {
    setOpenPackagePopup(false);
  };

  const [copy, setCopy] = useState(false);

  const goToFeedback = () => {
    window.open(
      `${process.env.NEXT_PUBLIC_REACT_APP_DOMAIN}/contact/provide-feedback`
    );
  };

  const goToApplicants = (jobTitle, jobId) => {
    window.location.href = `/employer/applicants/manage/active-jobs/${jobId}/${getFormattedJobTitle(
      jobTitle
    )}`;
  };

  return (
    <React.Fragment>
      <Head>
        <title>Employer MyHome - JobsHorn</title>
      </Head>
      <Container>
        <MainContent>
          <Grid container justifyContent="center">
            {openPackagePopup && (
              <BuyPackagePopup open={openPackagePopup} setOpen={onClosePopup} />
            )}
            <Grid item lg={8} xs={12}>
              <div className={classes["page_title"]}>
                <h1 className={classes["title"]}>
                  {`${empProfile ? empProfile.firstName : "Hi"} ${
                    empProfile ? empProfile.lastName : "Guest"
                  }, ready to hire !`}
                </h1>
                <span className={classes["sub_title"]}>
                  We are here to help, in your next great hire.
                </span>
              </div>
              {empProfile?.companyProfileTypeCode !== ACCOUNT_TYPES.ADMIN && (
                <Grid container alignItems="center" sx={{ p: "1% 0px" }}>
                  <Grid item xs={12} sx={{ textAlign: "center" }}>
                    <SectionHeading
                      sx={{
                        borderBottom: 0,
                        mb: 0,
                        pb: 0,
                        color: colors?.primaryDarkColor,
                      }}
                      variant="subtitle2"
                    >
                      Share your company
                    </SectionHeading>
                    {empProfile &&
                      empProfile?.companyProfileTypeCode !== undefined &&
                      empProfile?.companyProfileTypeCode !==
                        ACCOUNT_TYPES.ADMIN && (
                        <div className={classes["company-url"]}>
                          <a
                            href={`${process.env.NEXT_PUBLIC_REACT_APP_DOMAIN}/company/${empProfile?.companyNameUrl}/jobs`}
                            target="_blank"
                            id="jobs"
                            rel="noreferrer"
                          >
                            {`${process.env.NEXT_PUBLIC_REACT_APP_DOMAIN}/company/${empProfile?.companyNameUrl}/jobs`}
                          </a>
                        </div>
                      )}
                  </Grid>
                  <Grid item xs={12}>
                    <SocialShare
                      isSmallScreen={matches}
                      justifyContent={matches ? "flex-start" : "center"}
                      url={`${process.env.NEXT_PUBLIC_REACT_APP_DOMAIN}/company/${empProfile?.companyNameUrl}/jobs`}
                      title={`Check out ${empProfile?.companyName} Jobs`}
                      quote={`${process.env.NEXT_PUBLIC_REACT_APP_DOMAIN}/company/${empProfile?.companyNameUrl}/jobs`}
                      hashtags={"JobsHorn"}
                    />
                  </Grid>
                </Grid>
              )}
              {matches && !expanded && (
                <List>
                  <ListItem
                    secondaryAction={
                      <ExpandMore
                        expand={expanded}
                        onClick={handleExpandClick}
                        aria-expanded={expanded}
                        aria-label="show more"
                      >
                        <ExpandMoreIcon />
                      </ExpandMore>
                    }
                  >
                    <ListItemText
                      primaryTypographyProps={{
                        classes: { root: classes["expand-text"] },
                      }}
                      primary={"Show Package Details"}
                    />
                  </ListItem>
                </List>
              )}
              {!matches ? (
                <Grid container justifyContent="center">
                  <CustomCard
                    onClose={onClosePopup}
                    buyPackage={buyPackage}
                    data={packageData?.data}
                    open={openPackagePopup}
                  />
                </Grid>
              ) : (
                <Collapse in={expanded} timeout="auto" unmountOnExit>
                  <Grid container justifyContent="center">
                    <CustomCard
                      onClose={onClosePopup}
                      buyPackage={buyPackage}
                      data={packageData?.data}
                      open={openPackagePopup}
                    />
                  </Grid>
                </Collapse>
              )}
              {matches && expanded && (
                <List>
                  <ListItem
                    secondaryAction={
                      <ExpandMore
                        expand={expanded}
                        onClick={handleExpandClick}
                        aria-expanded={expanded}
                        aria-label="show less"
                      >
                        <ExpandMoreIcon />
                      </ExpandMore>
                    }
                  >
                    <ListItemText
                      primaryTypographyProps={{
                        classes: { root: classes["collapse-text"] },
                      }}
                      primary={"Hide Package Details"}
                    />
                  </ListItem>
                </List>
              )}
              <div className={classes["btns"]}>
                <CustomButton
                  height={4}
                  id="post-job"
                  startIcon={
                    <WrapperTooltip title="Post a job">
                      <img
                        className={classes["btn-image"]}
                        src={postAJobIcon}
                        alt=""
                      />
                    </WrapperTooltip>
                  }
                  variant="outlined"
                  onClick={() => {
                    window.location.href = "/employer/post-job";
                  }}
                >
                  Post a job
                </CustomButton>
                <CustomButton
                  height={4}
                  id="active-posted-job"
                  startIcon={
                    <WrapperTooltip title="Manage Jobs">
                      <img
                        className={classes["btn-image"]}
                        src={manageJobIcon}
                        alt=""
                      />
                    </WrapperTooltip>
                  }
                  variant="outlined"
                  onClick={() => {
                    window.location.href = "/employer/manage/active-jobs";
                  }}
                >
                  Manage Jobs
                </CustomButton>
                {/* <CustomButton
                height={4}
                id="talent-search"
                disabled
                startIcon={
                  <WrapperTooltip title="Talent Search">
                    <img
                      className={classes["btn-image"]}
                      src={talentSearchIcon}
                      alt=""
                    />
                  </WrapperTooltip>
                }
                variant="outlined"
              >
                Talent Search
              </CustomButton> */}
                <CustomButton
                  height={4}
                  id="company-active-jobs"
                  variant="outlined"
                  startIcon={
                    <WrapperTooltip title="Company jobs">
                      <CorporateFareOutlinedIcon className={classes["icon"]} />
                    </WrapperTooltip>
                  }
                  onClick={() => {
                    window.location.href = `/employer/company/active-jobs/${empProfile?.companyProfileTypeCode}`;
                  }}
                >
                  Company Jobs
                </CustomButton>
                {empProfile?.userTypeCode === USER_TYPES.EMPLOYER_ADMIN && (
                  <CustomButton
                    height={4}
                    id="company-information"
                    variant="outlined"
                    onClick={() => {
                      window.location.href = `/employer/company-information`;
                    }}
                  >
                    Manage Company Information
                  </CustomButton>
                )}
                {empProfile &&
                  empProfile.companyProfileTypeCode !== undefined &&
                  empProfile?.companyProfileTypeCode ===
                    ACCOUNT_TYPES.ADMIN && (
                    <CustomButton
                      height={4}
                      id="jh-admin-tools"
                      variant="outlined"
                      onClick={() => {
                        window.location.href = `/employer/admin/tools`;
                      }}
                    >
                      JH Admin Tools
                    </CustomButton>
                  )}
              </div>
              {empProfile &&
                empProfile?.companyProfileTypeCode !== undefined &&
                empProfile?.companyProfileTypeCode !==
                  ACCOUNT_TYPES.INDIVIDUAL && (
                  <div
                    className={`${classes["btns"]} ${
                      empProfile?.companyProfileTypeCode !==
                        ACCOUNT_TYPES.ADMIN &&
                      empProfile?.companyProfileTypeCode !==
                        ACCOUNT_TYPES.CORPORATE &&
                      empProfile?.companyProfileTypeCode !==
                        ACCOUNT_TYPES.ENTERPRISE
                        ? classes["company-btns"]
                        : ""
                    }`}
                  ></div>
                )}
              {empProfile &&
                empProfile?.companyProfileTypeCode !== undefined &&
                empProfile?.companyProfileTypeCode !== ACCOUNT_TYPES.ADMIN &&
                recentActivities?.length > 0 && (
                  <React.Fragment>
                    <SectionTitle sx={{ mb: 0, mt: 0 }} variant="h4">
                      Recent activities on jobs
                    </SectionTitle>
                    <List className={classes.list}>
                      {recentActivities.map((opt, i) => (
                        <div key={opt.jobId}>
                          <Paper
                            elevation={2}
                            square={false}
                            style={{ padding: "8px", margin: "12px 0" }}
                          >
                            <ListItem>
                              <ListItemText
                                disableTypography
                                primary={
                                  <CustomTypography>
                                    {`${opt.jobTitle}`}
                                  </CustomTypography>
                                }
                                secondary={
                                  !matches ? (
                                    <span>
                                      <Typography variant="subtitle2">
                                        {`Job Id: ${
                                          opt?.jobId
                                        } | Updated On: ${formatDate(
                                          opt.updatedTime
                                        )}`}
                                      </Typography>
                                    </span>
                                  ) : (
                                    <Grid container>
                                      <Grid item xs={12}>
                                        <Typography variant="subtitle2">
                                          Job Id: {opt?.jobId}
                                        </Typography>
                                      </Grid>
                                      <Grid item xs={12}>
                                        <Typography variant="subtitle2">
                                          Updated On:{" "}
                                          {formatDate(opt.updatedTime)}
                                        </Typography>
                                      </Grid>
                                      <Grid
                                        item
                                        xs={12}
                                        sx={{
                                          pt: "2%",
                                        }}
                                      >
                                        <Grid
                                          container
                                          justifyContent="flex-end"
                                        >
                                          <Grid item xs={7}>
                                            <ListRightAction
                                              onClick={() =>
                                                goToApplicants(
                                                  opt?.jobTitle,
                                                  opt?.jobId
                                                )
                                              }
                                              disabled={
                                                opt?.applicationsCount <= 0
                                              }
                                            >
                                              <IconButton
                                                disabled={
                                                  opt?.applicationsCount <= 0
                                                }
                                                size="small"
                                              >
                                                <PeopleIcon />
                                              </IconButton>
                                              <Typography>
                                                {opt?.applicationsCount}{" "}
                                                applicants
                                              </Typography>
                                            </ListRightAction>
                                          </Grid>
                                        </Grid>
                                      </Grid>
                                    </Grid>
                                  )
                                }
                              />
                              {!matches && (
                                <ListItemSecondaryAction>
                                  <ListRightAction
                                    onClick={() =>
                                      goToApplicants(opt?.jobTitle, opt?.jobId)
                                    }
                                    disabled={opt?.applicationsCount <= 0}
                                  >
                                    <WrapperTooltip title="Applicants">
                                      <span>
                                        <IconButton
                                          disabled={opt?.applicationsCount <= 0}
                                          size="small"
                                        >
                                          <PeopleIcon />
                                        </IconButton>
                                      </span>
                                    </WrapperTooltip>
                                    <Typography>
                                      {opt?.applicationsCount} applicants
                                    </Typography>
                                  </ListRightAction>
                                </ListItemSecondaryAction>
                              )}
                            </ListItem>
                          </Paper>
                        </div>
                      ))}
                    </List>
                  </React.Fragment>
                )}
              <ReferralContainer container justifyContent="space-between">
                <ReferralTextContainer>
                  <Typography
                    className={`${classes["recent_activities"]} ${classes["referral-link"]}`}
                  >
                    {`${PLACEHOLDERS.INVITE_YOUR_FRIENDS}`}
                  </Typography>
                  <Tooltip
                    title="Please copy and share this link with your Network & Friends"
                    placement={matches ? "bottom" : "right"}
                    arrow
                  >
                    <InfoOutlinedIcon
                      sx={{ ml: "0.25px" }}
                      fontSize="small"
                      className={classes["info-icon"]}
                    />
                  </Tooltip>
                </ReferralTextContainer>
                <Grid item xs={12}>
                  <ReferralLinkText>{referralLink}</ReferralLinkText>
                </Grid>
                <ReferralActions item xs={12}>
                  <Grid
                    alignItems="center"
                    container
                    justifyContent={!matches ? "flex-end" : "flex-start"}
                  >
                    <Grid
                      item
                      className={classes["referral-social-icon-btn"]}
                      xs={2}
                      lg="auto"
                    >
                      <Tooltip title="Share on LinkedIn" arrow placement="top">
                        <ShareIconButton
                          size="small"
                          onClick={() => handleSocialShare("linkedin")}
                        >
                          <LinkedInIcon />
                        </ShareIconButton>
                      </Tooltip>
                    </Grid>
                    <Grid
                      item
                      xs={2}
                      className={classes["referral-social-icon-btn"]}
                      lg="auto"
                    >
                      <Tooltip title="Share on Twitter" arrow placement="top">
                        <ShareIconButton
                          size="small"
                          onClick={() => handleSocialShare("twitter")}
                        >
                          <XIcon />
                        </ShareIconButton>
                      </Tooltip>
                    </Grid>
                    <Grid
                      item
                      className={classes["referral-social-icon-btn"]}
                      xs={2}
                      lg="auto"
                    >
                      <Tooltip title="Share on Facebook" arrow placement="top">
                        <ShareIconButton
                          size="small"
                          onClick={() => handleSocialShare("facebook")}
                        >
                          <FacebookIcon />
                        </ShareIconButton>
                      </Tooltip>
                    </Grid>
                    <Grid
                      item
                      className={classes["referral-social-icon-btn"]}
                      xs={2}
                      lg="auto"
                    >
                      <Tooltip title="Share via WhatsApp" arrow placement="top">
                        <ShareIconButton
                          size="small"
                          onClick={() => handleSocialShare("whatsapp")}
                        >
                          <WhatsAppIcon />
                        </ShareIconButton>
                      </Tooltip>
                    </Grid>
                    <Grid
                      item
                      className={classes["referral-social-icon-btn"]}
                      xs={2}
                      lg="auto"
                    >
                      <Tooltip title="Share via Telegram" arrow placement="top">
                        <ShareIconButton
                          size="small"
                          onClick={() => handleSocialShare("telegram")}
                        >
                          <TelegramIcon />
                        </ShareIconButton>
                      </Tooltip>
                    </Grid>
                    {/* <Grid item xs={2} lg="auto">
                    <Tooltip title="Share via Instagram" arrow placement="top">
                      <ShareIconButton
                        size="small"
                        onClick={() => handleSocialShare("instagram")}
                      >
                        <InstagramIcon />
                      </ShareIconButton>
                    </Tooltip>
                  </Grid> */}
                    <Grid
                      className={classes["referral-social-icon-btn"]}
                      item
                      xs={2}
                      lg="auto"
                    >
                      <Tooltip title="Share via Email" arrow placement="top">
                        <ShareIconButton
                          size="small"
                          onClick={() => handleSocialShare("email")}
                        >
                          <EmailIcon />
                        </ShareIconButton>
                      </Tooltip>
                    </Grid>
                    <Grid
                      className={classes["referral-social-icon-btn"]}
                      item
                      xs={2}
                      lg="auto"
                    >
                      <Tooltip
                        arrow
                        title={copy ? "Copied" : "Copy"}
                        placement="top"
                      >
                        <ShareIconButton
                          size="small"
                          id="refer-refcode-btns-emp"
                          onClick={handleCopyLink}
                        >
                          <ContentCopyOutlinedIcon />
                        </ShareIconButton>
                      </Tooltip>
                    </Grid>
                    <Grid
                      className={classes["referral-social-icon-btn"]}
                      item
                      xs={2}
                      lg="auto"
                    >
                      <Tooltip title="Generate QR" arrow placement="top">
                        <ShareIconButton
                          size="small"
                          onClick={() => {
                            const url = new URL(
                              "/employer/referral-share",
                              window.location.origin
                            );
                            url.searchParams.append("from", "myHome");
                            window.location.href = url.toString();
                          }}
                        >
                          <QrCodeScannerOutlinedIcon fontSize="small" />
                        </ShareIconButton>
                      </Tooltip>
                    </Grid>
                  </Grid>
                </ReferralActions>
              </ReferralContainer>
              <div className={classes["feedback-btns"]}>
                <CustomButton
                  height={!matches ? 2.5 : 4}
                  id="feedback-btns-emp"
                  variant="outlined"
                  onClick={goToFeedback}
                  sx={{ ml: 0 }}
                >
                  Provide your feedback
                </CustomButton>
                <CustomButton
                  sx={{ mr: 0 }}
                  height={!matches ? 2.5 : 4}
                  disabled
                  id="refer-recruiter-btns-emp"
                  variant="outlined"
                >
                  Refer a recruiter
                </CustomButton>
              </div>
            </Grid>
          </Grid>
        </MainContent>
      </Container>
    </React.Fragment>
  );
};

export default EmpHomePage;
